uta_tools/main.py

```diff
@@ -220,3 +225,36 @@ async def get_mapped_mane_data(
         warnings=warnings,
         service_meta=uta_tools.service_meta()
     )
+
+
+@app.get(
+    f"/{SERVICE_NAME}/download_sequence",
+    summary="Get sequence for ID",
+    response_description=RESP_DESCR,
+    description="Given a known accession identifier, retrieve sequence data and return"
+                "as a FASTA file",
+    response_class=FileResponse
+)
+async def get_sequence(
+    background_tasks: BackgroundTasks,
+    sequence_id: str = Query(..., description="ID of sequence to retrieve"),
+) -> FileResponse:
+    """Get sequence for requested sequence ID.
+    :param sequence_id: accession ID, sans namespace
+    :param background_tasks: Starlette background tasks object. Use to clean up
+        tempfile after get method returns.
+    :return: FASTA file if successful, or 404 if unable to find matching resource
+    """
+    _, path = tempfile.mkstemp(suffix=".fasta")
+    try:
+        uta_tools.get_fasta_file(sequence_id, Path(path))
+    except KeyError:
+        raise HTTPException(
+            status_code=404,
+            detail="No sequence available for requested identifier"
+        )
+    background_tasks.add_task(
+        lambda p: os.unlink(p),
+        path
+    )
+    return FileResponse(path)
```

Comment:
This doesn't necessarily need to live in UTA Tools, but I wrote it to make sure that the file creation implementation would work with a FastAPI endpoint and left it in by default (I'll have to copy/paste to use in the curation page).